src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -250,18 +250,32 @@ class MarathonSchedulerService @Inject() (
 
   override def onElected(abdicateCmd: ExceptionalCommand[JoinException]): Unit = {
     log.info("Elected (Leader Interface)")
-
+    var migrationComplete = false
     try {
       //execute tasks, only the leader is allowed to
       migration.migrate()
+      migrationComplete = true
 
       // We have been elected. Thus, elect leadership with the abdication command.
       electLeadership(Some(abdicateCmd))
+
+      // We successfully took over leadership. Time to reset backoff
+      resetOfferLeadershipBackOff
     }
     catch {
-      case e: Exception =>
+      case e: scala.Exception =>
```

Comment:
Oops... you're right... there is a class `Exception` in the `mesosphere.marathon` namespace.